src/scheduler/scheduler.directive.js

```diff
@@ -0,0 +1,101 @@
+(function(){
+    'use strict';
+
+    angular.module('motech-scheduler')
+        .directive('collapser', function($compile) {
+        return {
+            restrict: 'C',
+            scope: {
+                collapser: '='
+            },
+            link: function (scope, element, attrs) {
+                element.bind('click', function() {
+                    var target = $(scope.collapser),
+                        shown = target.hasClass('in');
+
+                    if (shown) {
+                        $(scope.collapser).collapse('toggle');
+                        element.find('.fa-caret-down').removeClass("fa-caret-down").addClass("fa-caret-right");
+                        target.find('.collapse').collapse('hide');
+                        target.find('.fa-caret-down').removeClass("fa-caret-down").addClass("fa-caret-right");
+                    } else {
+                        element.find('.fa-caret-right').removeClass("fa-caret-right").addClass("fa-caret-down");
+                        $(scope.collapser).collapse('toggle');
+                    }
+                });
+            }
+        };
+    });
+
+    angular.module('motech-scheduler')
+        .directive("mtInvalid", function() {
+        return {
```

Comment:
Lets split it into its own file (one object per file) // if this is a very common pattern, would it be worth adding into /common/forms? I'd love to not be reinventing the wheel in separate modules...

IMHO, adding css classes at this level should either be one in a global manner... or in the actual mark-up

This might be something we should leave and create a new ticket for if we want to make it a global method (I haven't noticed it in other modules)